src/System.Linq/tests/MinMaxTests.cs

```diff
@@ -0,0 +1,373 @@
+﻿using System.Collections.Generic;
+using Xunit;
+
+namespace System.Linq.Tests
+{
+    public class MinMaxTests
+    {
+        [Fact]
+        public void MinInt()
+        {
+            var one = Enumerable.Range(1, 10).ToArray();
+            var minusTen = new[] { -1, -10, 10, 200, 1000 };
+            var hundred = new[] { 3000, 100, 200, 1000 };
+            Assert.Equal(1, one.Min());
+            Assert.Equal(-10, minusTen.Min());
+            Assert.Equal(100, hundred.Min());
+            Assert.Throws<InvalidOperationException>(() => Enumerable.Empty<int>().Min());
```

Comment:
Can you add more boundary cases, e.g. where the array includes Int32.MinValue, where there's a single element in the array?  And could you add some tests on non-array sources?